### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by visse0001**
Ready to merge.
I decided to use djangorestframework-jwt
This package provides JSON Web Token Authentication

To check how the tokens work, follow the instructions in README :)
You can also check how the tokens work by logging in as admin by hitting http://localhost:8000/admin in your browser and then hitting private endpoints e.g. http://localhost:8000/api/users/students/

- Added to requirements djangorestframework-jwt, so you should build and up docker-compose again -> `docker-compose up --build`, then populate data by `make load_data`
- Added to env_config/local.env CORS_ORIGIN_WHITELIST=localhost:3000 -> the domain for react app
- Added to settings REST_FRAMEWORK
- Added two serializes: CustomUserSerializer, CustomUserSerializerWithToken for handling signups
- Added new urls and views

You may copy and paste your token here https://jwt.io/ and see header, payload and signature, if you want :)
More info is available in commits history.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -61,18 +61,27 @@ Go to http://localhost:8000/admin and create a new user.
 
 ## Access API
 
-### Examples for students  
+### Auth
+
+Go to http://localhost:8000/api/users/register
```
Comment (tgbdc7) [inline README.md]: It should be http://localhost:8000/api/users/register/ 
Missing `/` at the end

File: README.md [hunk 2]
```diff
@@ -61,18 +61,27 @@ Go to http://localhost:8000/admin and create a new user.
 
 ## Access API
 
-### Examples for students  
+### Auth
+
```
Comment (tgbdc7) [inline README.md:65]: Maybe worth adding that use Postman app

File: README.md [hunk 3]
```diff
@@ -61,18 +61,27 @@ Go to http://localhost:8000/admin and create a new user.
...
+As you can see below, you can fill the body form-data with info:   
+
+![](https://github.com/zetonteam/zeton_django/blob/auth/images/postman_register_01.png?raw=true)  
+
+After that you can hit http://localhost:8000/api/users/students/ with GET method and put into Headers key and value.
+
+Then you have access to private endpoint.   
+
+![](https://github.com/zetonteam/zeton_django/blob/auth/images/postman_register_02.png?raw=true)  
+
```
Comment (tgbdc7) [inline README.md:80]: Could you explain why we must paste prefix `JWT` in token When we send a post request?

**Review by tgbdc7 — APPROVED** [review]
good job :ok_woman:

File: README.md (hunk 3 above)
Comment (visse0001) [reply README.md:80]: Get request. not post.

This is similar to how you put a token in Postman using the Authorization -> and selecting Type -> here we have to select 'JWT', but I don't see 'JWT' on the list, so I passed it to the Headers.

Bearer Token (or Basic) doesn't work here, I don't know why, if I find a moment I'll read about it.

File: README.md (hunk 2 above)
Comment (visse0001) [reply README.md:65]: done

File: README.md (hunk 1 above)
Comment (visse0001) [reply README.md]: done

**Pull request merged**